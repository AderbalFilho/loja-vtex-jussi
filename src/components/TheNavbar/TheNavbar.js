import React from 'react';
import PropTypes from 'prop-types';

import { JLink } from '@ui';

import './TheNavbar.scss';

function TheNavbar({ links }) {
  return (
    <nav className="the-navbar">
      {links &&
        links.map((link) => (
          <JLink classes="the-navbar__link" key={link.label} href={link.url}>
            {link.label}
          </JLink>
        ))}
    </nav>
  );
}

TheNavbar.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string,
      label: PropTypes.string,
    })
  ).isRequired,
};

export default TheNavbar;
