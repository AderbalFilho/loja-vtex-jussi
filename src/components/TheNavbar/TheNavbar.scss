.the-navbar__link {
  margin: 0 0 0 32px;
}
