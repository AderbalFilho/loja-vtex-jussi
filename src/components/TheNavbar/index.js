import TheNavbar from './TheNavbar';

export default TheNavbar;
