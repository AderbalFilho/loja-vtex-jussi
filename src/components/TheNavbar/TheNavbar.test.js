import React from 'react';
import TestRenderer from 'react-test-renderer';

import TheNavbar from './TheNavbar';

describe('TheNavbar', () => {
  it('matches snapshot', () => {
    const links = [
      { url: '#', label: 'Nossas soluções' },
      { url: '#', label: 'Conheça a Jüssi' },
    ];
    const testRenderer = TestRenderer.create(<TheNavbar links={links} />);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
