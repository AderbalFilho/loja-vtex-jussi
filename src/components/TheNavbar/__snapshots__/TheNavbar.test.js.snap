// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNavbar matches snapshot 1`] = `
<nav
  className="the-navbar"
>
  <a
    className="j-link the-navbar__link"
    href="#"
  >
    Nossas soluções
  </a>
  <a
    className="j-link the-navbar__link"
    href="#"
  >
    Conheça a Jüssi
  </a>
</nav>
`;
